ecl/eclcc/eclcc.hpp

```diff
@@ -0,0 +1,127 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef ECLCC_HPP
+#define ECLCC_HPP
+
+// The following is used by eclcc and eclcmd (ecl <command>).
+// The primary intention is for eclcc and only a selection are also used by eclcmd for its eclcc options.
+// The markers '!' and '?', present in either the first or first two columns, indicate that lines inclusion.
+// e.g. '?' indicates inclusion in both eclcmd and eclcc).
+//      '?!' indicates inclusion in the verbose usage for both.
+//      '!' indicates inclusion in the verbsoe usage for eclcc only.
+//      None of the above indicates inclusion in eclcc only.
+
+const char * const helpText[] = {
+    "",
+    "Usage:",
+    "    eclcc <options> queryfile.ecl",
+    "",
+    "General options:",
+    "?   -I <path>     Add path to locations to search for ecl imports",
+    "?   -L <path>     Add path to locations to search for system libraries",
+    "    -o <file>     Specify name of output file (default a.out if linking to",
+    "                  executable, or stdout)",
+    "?   -manifest     Specify path to manifest file listing resources to add",
+    "    -foption[=value] Set an ecl option (#option)",
+    "    -main <ref>   Compile definition <ref> from the source collection",
+    "    -syntax       Perform a syntax check of the ECL",
+    "    -platform=hthor Generate code for hthor executable (default)",
+    "    -platform=roxie Generate code for roxie cluster",
+    "    -platform=thor  Generate code for thor cluster",
+    "",
+    "Output control options",
+    "    -E            Output preprocessed ECL in xml archive form",
+    "!   -M            Output meta information for the ecl files",
+    "!   -Md           Output dependency information",
+    "!   -Me           eclcc should evaluate supplied ecl code rather than generating a workunit",
+    "    -q            Save ECL query text as part of workunit",
+    "    -wu           Only generate workunit information as xml file",
+    "",
+    "c++ options",
```

Comment:
Should do when we add more of the other options in that section, along with the other corresponding section titles e.g. "Other options:".
For now, I wanted it it conform as much as it could to the original ecl run usage format.